api/incidents/views.py

```diff
@@ -128,3 +128,33 @@ def retrieve(self, request, pk=None):
         return JsonResponse({"status": 200,
                              "data": serializer.data},
                             status=200)
+
+    def update(self, request, **kwargs):
+        """
+        update:
+        update a redflag.
+        """
+        try:
+            instance = self.get_object()
+
+            if int(instance.createdBy) != request.user.id:
+                return JsonResponse({"status": 403,
+                                     "message": "You can not update a redflag you do not own"},
+                                    status=403)
+
+            if instance.status.lower() == 'resolved':
```

Comment:
The status can either be resolved, under investigation, draft etc. Checking only for the case for resolved means if the record is under investigation, it can still be updated. This should not be the case.